Description: Fix limit readline output to buffer size
Author: Juho Heikkinen <juho.heikkinen@f-secure.com>
Bug: https://github.com/unbit/uwsgi/issues/1412
Bug-Debian: https://bugs.debian.org/846362
Forwarded: yes
Last-Update: 2018-10-02
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/core/reader.c
+++ b/core/reader.c
@@ -283,7 +283,7 @@
                 }
 
 	// no line found, let's return all
-        *rlen = wsgi_req->post_readline_size - wsgi_req->post_readline_pos;
+        *rlen = wsgi_req->post_readline_watermark - wsgi_req->post_readline_pos;
         char *buf = wsgi_req->post_readline_buf + wsgi_req->post_readline_pos;
 	wsgi_req->post_readline_pos = 0;
 	return buf;
